packages/back-end/src/api/segments/listSegments.ts

```diff
@@ -12,7 +9,7 @@ import { listSegmentsValidator } from "../../validators/openapi";
 
 export const listSegments = createApiRequestHandler(listSegmentsValidator)(
   async (req): Promise<ListSegmentsResponse> => {
-    const segments = await findSegmentsByOrganization(req.organization.id);
+    const segments = await req.context.models.segments.getAll();
 
     // TODO: Move sorting/limiting to the database query for better performance
```

Comment:
TODO: Now that we're using the BaseModel, do the filtering/limiting there via the use of the find() method, with the limit/sort query.